### PR TITLE
[CLIENT/Team4] 2 out of 4

### PR DESCRIPTION
**Pull request by akowal3**
# Summary

Do not merge before #384 

- Added trust datastructure and helper functions
- gamestate is now passed to judge/speaker (major bugfix)

### CODE REVIEW

**Comment by lhl2617** [comment]
converting to draft--waiting for #384

File: internal/clients/team4/client.go [hunk 1]
```diff
@@ -4,10 +4,12 @@ package team4
 import (
 	"math"
 	"sort"
+	"testing"
```
Comment (lhl2617) [inline internal/clients/team4/client.go]: what is this doiong here

**Review by NGovani — APPROVED** [review]
LGTM

File: internal/clients/team4/client_test.go [hunk 2]
```diff
@@ -7,12 +7,68 @@ import (
...
+
 	winner := j.DecideNextPresident(shared.Team1)
 
 	if winner != shared.Team1 {
 		t.Errorf("Got wrong judge pointer. Winner is %v", winner)
 	}
-
 }
+
+// func TestUpdateTrustFromHistory(t *testing.T) {
```
Comment (lhl2617) [inline internal/clients/team4/client_test.go:22]: image

Comment (Spruudis) [reply internal/clients/team4/client_test.go:22]: I'll ask Andrzej tmrw when he wakes up

File: internal/clients/team4/client.go (hunk 1 above)
Comment (akowal3) [reply internal/clients/team4/client.go]: pointed out in #386

File: internal/clients/team4/client_test.go (hunk 2 above)
Comment (akowal3) [reply internal/clients/team4/client_test.go:22]: I used it to print out the values to look at them ¯\\\_(ツ)_/¯. Will remove

**Pull request merged**